### ISSUE TITLE
Trip planner slowdown

### ISSUE THREAD

**Issue by jmfield2** [CUTR-at-USF/usf-mobullity#135]
Over the past week, JB noticed that the trip planner was taking much more time than normal to create a plan.  The service did not seem to be exhibiting any clear signs indicating an issue when I connected, and the memory available on the server was not unusual.

Restarting the mobullityOTP service seems to have cleared the problem up for now.

Some thoughts:
- Last month I added -Xmx2G parameter to allow OTP to use more ram.  Perhaps this was causing an issue.
- Network-related issues arising from the reIP of mobullity2/3 (seems unlikely)

**Comment by jmfield2** [comment]
Over the past week I've done some more troubleshooting on this including the following:
- Testing without any Graph updaters (Bus/Bike) (by removing Graph.properties temporarily)
- Testing long distance mode (-l cmdline)
- Testing on a different machine (my laptop) with the same tampa graph (using OpenJDK 8)
- Updating to the latest 64-bit Java8 from Oracle

Nothing helped this except the last two tries, although the long distance mode did offer some relief and did speed up routes initiallly I don't believe this will have any practical application for us since we like providing multiple route options.

After trying to isolate the issue by changing the certain parameters a bit, I thought there might be some strangeness going on with Java or Windows because in each case the VM working set (not commit charge) would reduce from around 800MB on start to 600MB or below overnight and yield the erroneous path search timeout errors that manifest in the long trip planning delays that we were seeing.

So, I tried leaving the server running for longer than a day or so on a different machine.(Linux, too) and found the issue to be reduced dramatically or eliminated. Of course, I was also using a different version of Java (OpenJDK 8).

When I tried this on mobullity by removing all old JREs and installing the latest JRE8 from Oracle, as of now the problem has not resurfaced like before.

**Comment by barbeau** [comment]
@jmfield2 Thanks for troubleshooting this.  I know Oracle is pushing hard to get everyone off Java 6 & 7, so it could definitely be a JDK/JRE bug.  If JRE 8 seems to solve the issue and nothing else falls apart, let's go with it.

**Comment by barbeau** [comment]
@jmfield2 Do you think we can close out this issue for now?  I haven't seen the problem in a while, and it's possible one of the above fixes worked.

**Comment by jmfield2** [comment]
I still see slow trip plans on mobullity, but I don't think I've seen them on maps so far.  So, it seems like it is related either to usage (since most people are using maps, and not mobullity), or some configuration on mobullity that is not present on production.

I checked again and noticed that there are additional scheduled tasks on mobullity (chrome/google updater) - I disabled them.  So, if we don't see any issues in a few days to a week now I would consider this closed
